### ISSUE TITLE
Rehearsal matrix stage never converges: ExperimentVariant.name forbids underscores + engine template teaches single-variant/empty-overrides shape

### ISSUE THREAD

**Issue by OffensiveGeneric** [ccny-glasslab/glasslab-cluster-config#474]
## Summary

The rehearsal's experiment-matrix stage cannot converge. Run `16191d12f7f546d28baa75679c7d3946` reached `BEAKER_REVISING`, was rejected 3 times, and escalated to `methodology.human_resolution_requested` -> `PAUSED` (harness correctly refused to auto-resume). With `temperature: 0` this is a deterministic fixed point, so it can never converge by retrying.

## Deterministic root cause

The orchestrator policy denied the matrix with 3 schema errors:

```
experiment matrix schema validation failed: 3 validation errors for ExperimentMatrix
variants.0.name  String should match pattern '^[a-z0-9][a-z0-9-]{0,62}$' [input_value='gradient_clipping']
variants.1.name  ... [input_value='weight_decay']
variants.2.name  ... [input_value='learning_rate_scheduling']
```

`ExperimentVariant.name` (`services/research-orchestrator/app/schemas.py:402`) is `^[a-z0-9][a-z0-9-]{0,62}$` — **hyphens only, underscores forbidden** — while **every sibling `name` field in the same file allows underscores** (`^[a-z0-9][a-z0-9_-]{0,62}$` at lines 218, 735, 756, 781). Beaker naturally names methods `gradient_clipping`, `weight_decay`, `learning_rate_scheduling`, so the matrix is DENIED at policy validation every time.

Nothing downstream requires kebab-case:
- `JobRecord.variant_name` / `evidence.variant_name` are unconstrained `str` (schemas.py:502, 707).
- `matrix.py` uses `variant.name` only as a `'variant'` value in the job config and as an input to `uuid5(...)` for `orchestrator_job_id`.
- `cluster.py:80` builds `f'fake-{spec.variant_name}-{spec.seed}'` (fake executor only).

## Contributing defect: the engine teaches the shape both reviews reject

`engine.py::_matrix_action_template` hands the model this as the "Required submit_experiment_matrix action shape":

```python
'variants': [{'name': 'candidate', 'overrides': {}}],
```

i.e. **one variant with empty `overrides`**. Honeydew's review rejected exactly that:

> "3 seeds x 1 variant = 3 jobs. Matrix is methodologically unacceptable because variants array has empty overrides, failing to compare 3 distinct methods"

So the engine both (a) rejects the names the model produces and (b) demonstrates the single-empty-variant shape the reviewer rejects.

## Impact

Blocks end-to-end completion at the matrix stage for any contract with a `comparison` methodology requirement (`minimum_distinct_values >= 2`) — which is the normal case per `METHODOLOGY_REQUIREMENTS_GUIDANCE`.

## Acceptance

- A `comparison` methodology requirement produces a deterministically-convergent matrix: one variant per required distinct method, each with distinct **non-empty** `overrides`, and names that validate.
- The variant-name rule is stated explicitly to the agent (the pattern must not be discoverable only by trial-and-error rejection).
- Variant names using underscores validate (consistent with sibling fields).
- Tests: snake_case variant names validate; the generated action template/prompt for a `comparison` requirement contains `minimum_distinct_values` distinct variants with distinct non-empty overrides and names matching the documented pattern.
